### PR TITLE
Migrate team_list and team_row to typescript

### PR DESCRIPTION
**Pull request by vanya829**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/13489

### CODE REVIEW

**Review by crspeller — APPROVED** [review]

File: components/admin_console/team_channel_settings/team/list/__snapshots__/team_list.test.tsx.snap [hunk 1]
```diff
@@ -134,6 +134,7 @@ exports[`admin_console/team_channel_settings/team/TeamList should match snapshot
                 >
                   DN
                 </b>
+
```
Comment (nevyangelova) [inline components/admin_console/team_channel_settings/team/list/__snapshots__/team_list.test.tsx.snap:137]: seems like these spaces are coming from a diff formatter settings can we remove them?

**Review by nevyangelova — APPROVED** [review]
Otherwise LGTM 👍

**Review by josephbaylon — APPROVED** [review]
LGTM. Verified unit tests exist and also ran `search_teams_spec` e2e against the changes.

**Comment by hanzei** [comment]
Thank you @vanya829 :+1:

**Pull request merged**